src/pages/VerifySettingsPage/VerifySettingsPage.tsx

```diff
@@ -0,0 +1,70 @@
+import { useState, useCallback } from "react";
+import { useDeskproAppEvents, useDeskproAppClient } from "@deskpro/app-sdk";
+import { getAccounts } from "../../api/api";
+import { VerifySettings } from "../../components/VerifySettings";
+import type { FC } from "react";
+import type { Maybe, Settings } from "../../types";
+import type { IAccount } from "../../api/types";
+
+const getError = (err: Maybe<Error>): string => {
+  const defaultError = "Failed to connect to Wrike, settings seem to be invalid";
+  let error;
+
+  if (!err) {
+    error = defaultError;
+  } else {
+    try {
+      const parsedError = JSON.parse(err.message);
+      const parsedMessage = JSON.parse(parsedError.message);
+      error = parsedMessage.errorDescription || defaultError;
+    } catch (e) {
+      error = defaultError;
+    }
+  }
+
+  return error;
+};
```

Comment:
Minor: This can be simplified.
```
const getError = (err: Maybe<Error>): string => {
  const defaultError = "Failed to connect to Wrike, settings seem to be invalid";
  try {
      const parsedError = JSON.parse(err.message);
      const parsedMessage = JSON.parse(parsedError.message);
      if (parsedMessage.errorDescription) {
          return parsedMessage.errorDescription;
      }
   } catch (e) {
     // do nothing
   }

  return defaultError;
}
```

Reply:
simplified